src/agents/coordinator_agent.py

```diff
@@ -0,0 +1,172 @@
+        # {user_jid: {parkings: [parking_available, ...], destination: (x, y)}}
+        self._per_user_data = {}
+
+    def _prepare_check_offers_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "query-ref")
+        template.set_metadata("action", "check-offers")
+        return template
+
+    def _prepare_make_reservation_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "request")
+        template.set_metadata("action", "make-reservation")
+        return template
+
+    def _prepare_modify_reservation_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "request")
+        template.set_metadata("action", "modify-reservation")
+        return template
+
+    def _prepare_parking_availability_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "inform")
+        template.set_metadata("action", "parking-availability")
+        return template
+
+    def _prepare_reservation_response_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "inform")
+        template.set_metadata("action", "reservation-response")
+        return template
+
+    def _calculate_distance(self, x_user, y_user, x_parking, y_parking):
+        return ((x_user - x_parking) ** 2 + (y_user - y_parking) ** 2) ** 0.5
+
+    def _consolidate_offers_per_user(self, user) -> ConsolidatedOffers:
+        dest_x, dest_y = self._per_user_data[user]["destination"]
+        free_parkings = [parking for parking in self._per_user_data[user]["parkings"] if parking.available]
+        offers = []
+        for parking in free_parkings:
+            offer = Offer(parking_id=parking.parking_id, price=parking.parking_price,
+                          distance=self._calculate_distance(dest_x, dest_y, parking.parking_x, parking.parking_y))
+            offers.append(offer)
+        offers = sorted(offers, key=lambda x: x.distance)
+        offers = offers[:3]
+        return ConsolidatedOffers(offers=offers)
+
+    class CheckParkingOffers(CyclicBehaviour):
+        """Behaviour for checking parking offers"""
+
+        def _check_if_request_is_within_region(self, request_x, request_y):
+            return self.agent._x_min <= request_x < self.agent._x_max and self.agent._y_min <= request_y < self.agent._y_max
+
+        async def run(self):
+            msg = await self.receive(timeout=1)
+            if msg:
+                check_offers_message = CheckOffers(**json.loads(msg.body))
+                self.agent._per_user_data[check_offers_message.sender] = {
```

Comment:
Let's move this maybe after the if_request_within_region check? If the position is incorrect, do you think we should still keep the information about that user's message?